numba/np/arraymath.py

```diff
@@ -4836,3 +4842,162 @@ def impl(filt, trim='fb'):
+        # b) the second condition is true (i.e. `ar1` or `ar2` may contain
+        #    arbitrary objects), since then sorting is not guaranteed to work
+        if len(ar2) < 10 * len(ar1) ** 0.145:
+            if invert:
+                mask = np.ones(len(ar1), dtype=np.bool_)
+                for a in ar2:
+                    mask &= (ar1 != a)
+            else:
+                mask = np.zeros(len(ar1), dtype=np.bool_)
+                for a in ar2:
+                    mask |= (ar1 == a)
+            return mask
+
+        # Otherwise use sorting
+        if not assume_unique:
+            # Equivalent to ar1, inv_idx = np.unique(ar1, return_inverse=True)
+            # https://github.com/numpy/numpy/blob/
+            # 03b62604eead0f7d279a5a4c094743eb29647368/
+            # numpy/lib/arraysetops.py#L358C8-L358C8
+            order1 = np.argsort(ar1)
+            aux = ar1[order1]
+            mask = np.empty(aux.shape, dtype=np.bool_)
+            mask[:1] = True
+            mask[1:] = aux[1:] != aux[:-1]
+            ar1 = aux[mask]
+            imask = np.cumsum(mask) - 1
+            inv_idx = np.empty(mask.shape, dtype=np.intp)
+            inv_idx[order1] = imask
+            ar2 = np.unique(ar2)
+
+        ar = np.concatenate((ar1, ar2))
+        # We need this to be a stable sort, so always use 'mergesort'
+        # here. The values from the first array should always come before
+        # the values from the second array.
+        order = ar.argsort(kind='mergesort')
+        sar = ar[order]
+        flag = np.empty(sar.size, np.bool_)
+        if invert:
+            flag[:-1] = (sar[1:] != sar[:-1])
+        else:
+            flag[:-1] = (sar[1:] == sar[:-1])
+        flag[-1:] = invert
+        ret = np.empty(ar.shape, dtype=np.bool_)
+        ret[order] = flag
+
+        # return ret[:len(ar1)]
+        if assume_unique:
+            return ret[:len(ar1)]
+        else:
+            return ret[inv_idx]
+
+    return np_in1d_impl
+
+
+@overload(np.isin)
+def jit_np_isin(element, test_elements, assume_unique=False, invert=False,
+                kind="sort"):
+    if not (type_can_asarray(element) or type_can_asarray(test_elements)):
+        raise TypingError('in1d: first two args must be array-like')
+    if not (isinstance(assume_unique, (types.Boolean, bool))):
+        raise TypingError('in1d: Argument "assume_unique" must be boolean')
+    if not (isinstance(invert, (types.Boolean, bool))):
+        raise TypingError('in1d: Argument "invert" must be boolean')
+    if not (isinstance(kind, (types.UnicodeType, str))):
+        raise TypingError('in1d: Argument "kind" must be string')
+
+    # https://github.com/numpy/numpy/blob/
+    # 03b62604eead0f7d279a5a4c094743eb29647368/numpy/lib/arraysetops.py#L889
+    def np_isin_impl(element, test_elements, assume_unique=False, invert=False,
+                     kind="sort"):
+        if kind != "sort":
+            raise ValueError('Only kind="sort" is supported')
```

Comment:
Move this check to run during typing

Reply:
How would I do this? Using a check for ``Literal[str]("sort")``?

Reply:
Yes! But actually, let's keep it like this.